File: test-programs/erlang/input.erl
%% -*- erlang -*-

%% Main function that gets executed when the script runs
main(_Args) ->
    %% Read and print input from stdin
    loop().

%% Loop to continuously read lines from stdin until eof
loop() ->
    case io:get_line('') of
        eof -> ok;  %% End of input
        Line -> 
            io:format("~s", [Line]),  %% Print the line
            loop()  %% Continue the loop
    end.

File: test-programs/rust/input.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    for line in stdin.lock().lines() {
        match line {
            Ok(l) => println!("{}", l),
            Err(e) => eprintln!("Error: {}", e),
        }
    }
}
